### ISSUE TITLE
Don't use chroot, use su

### ISSUE THREAD

**Issue by hboetes** [elastic/logstash#4294]
I noticed this error message while debugging logstash initscript On redhat 6 x86_64:

``` bash
[root@con-logger]# grep logstash /etc/security/limits.conf
logstash soft nofile 16384
logstash hard nofile 65536
[root@con-logger]# nice -n 19 chroot --userspec logstash:logstash --groups logstash /
bash: /root/.bashrc: Permission denied
bash-4.1$ ulimit -n
1024
bash-4.1$ ulimit -n 16384
bash: ulimit: open files: cannot modify limit: Operation not permitted
bash-4.1$ exit
[root@con-logger]# nice -n 19 su - logstash
-bash-4.1$ ulimit -n
16384
-bash-4.1$ ulimit -n 65536
-bash-4.1$ ulimit -n
65536
```

And this is a the code in question, from `/etc/init.d/logstash`:

``` bash
 nice -n ${LS_NICE} chroot --userspec $LS_USER:$LS_GROUP $EXTRA_GROUPS / sh -c "                         
    cd $LS_HOME                                                                                           
    ulimit -n ${LS_OPEN_FILES}                                                                            
    exec \"$program\" $args                                                                               
  " > "${LS_LOG_DIR}/$name.stdout" 2> "${LS_LOG_DIR}/$name.err" &
```

Therefore my recommendations:
- Use `su(1)` instead
- Set the logstash user homedir to `$LS_HOME`
  
  `# grep logstash /etc/passwd
  logstash:x:494:495:logstash:/var/lib/logstash:/bin/bash`
- Add the proper limits for logstash to `/etc/security/limits.d/50-logstash`

**Comment by hrchu** [comment]
I have the same issue in ubuntu 10.04

My fix is :nice -n ${LS_NICE} su -s /bin/sh $LS_USER -c "

**Comment by jordansissel** [comment]
I believe this is fixed in Logstash 5.0 for the reported linux distros in this ticket:

* Ubuntu 10.04 uses Upstart, and Logstash will install an upstart job upon package install if it detects Upstart is available. This resolves the problem by removing the sysv script this bug is about.
* CentOS 6 also uses Upstart.

**Comment by jordansissel** [comment]
Please reopen or open a new ticket if you still have this issue in Logstash 5.0
